src/App.tsx

```diff
@@ -1,56 +1,76 @@
-import React from 'react';
+import React, {
+  useMemo, useState, useEffect, useCallback,
+} from 'react';
 import './App.scss';
+import classNames from 'classnames';
 import { peopleFromServer } from './data/people';
+import { PeopleList } from './components/PeopleList';
+import { Person } from './types/Person';
+import { debounce } from './services/debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [person, setPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+
+  const debouncedQuery
+  = useCallback(debounce(value => setQuery(value), 1000), []);
+
+  useEffect(() => {
+    debouncedQuery(query);
+  }, [query, debouncedQuery]);
+
+  const filteredPerson = useMemo(() => {
+    return peopleFromServer
+      .filter(human => human.name.toLowerCase().includes(query.toLowerCase()));
+  }, [query]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = event.target.value;
+
+    setQuery(inputValue);
+  };
```

Comment:
```suggestion
  const [query, setQuery] = useState('');
  const [appliedQuery, setAppliedQuery] = useState('');
  const [isFocused, setIsFocused] = useState(false);

  const debouncedQuery = useCallback(debounce(setAppliedQuery, 1000), []);

  const filteredPerson = useMemo(() => {
    return peopleFromServer
      .filter(human => human.name.toLowerCase().includes(appliedQuery.toLowerCase()));
  }, [appliedQuery]);

  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const inputValue = event.target.value;

    setQuery(inputValue);
    debouncedQuery(inputValue);
  };
```
debounce still not works.
should be such changes